### ISSUE TITLE
LinearProgressIndicator and CircularProgressIndicator Documentation.

### ISSUE THREAD

**Issue by SushantChandla** [flutter/flutter#87920]
LinearProgressIndicator and CircularProgressIndicator documentation does not specify that value parameter should be between 0 and 1.

## Steps to Reproduce
Documentation issue.

![LinearProgressIndicator](https://i.imgur.com/JUsetUU.png)
it is similar for CircularProgressIndicator.

<details>
  <summary>Logs</summary>

```
❯ flutter doctor -v
[✓] Flutter (Channel master, 2.5.0-2.0.pre.128, on macOS 11.5.1 20G80
    darwin-arm, locale en-IN)
    • Flutter version 2.5.0-2.0.pre.128 at /Users/sushantchandla/tools/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 08320a8dc4 (6 days ago), 2021-08-03 01:05:02 -0400
    • Engine revision eb1d87d805
    • Dart version 2.14.0 (build 2.14.0-372.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/sushantchandla/Library/Android/Sdk
    • Platform android-31, build-tools 31.0.0
    • ANDROID_HOME = /Users/sushantchandla/Library/Android/Sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] VS Code (version 1.59.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.25.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 11.5.1 20G80 darwin-arm
    • Chrome (web)    • chrome • web-javascript • Google Chrome 92.0.4515.131

• No issues found!
```
</details>

I would like to make a pr fixing this : )

**Comment by darshankawar** [comment]
For LinearProgressIndicator, it does seem to mention the value, as below:

<img width="929" alt="Screenshot 2021-08-09 at 5 20 08 PM" src="https://user-images.githubusercontent.com/67046386/128701875-ab2a12e8-8439-433d-9c7d-4b171ac60df8.png">

https://api.flutter.dev/flutter/material/LinearProgressIndicator-class.html

For CircularProgressIndicator, it does seem not to have mentioned the range though.

Labeling it as applicable.

**Comment by nt4f04uNd** [comment]
This link is for the constructor. Here's a class doc link, which does mention that https://api.flutter.dev/flutter/material/CircularProgressIndicator-class.html

**Comment by nt4f04uNd** [comment]
The https://api.flutter.dev/flutter/material/RefreshProgressIndicator-class.html doc comment though would probably benefit from having the same section

**Comment by creativecreatorormaybenot** [comment]
I will open a PR to fix this in a bit 👍

**Comment by creativecreatorormaybenot** [comment]
@SushantChandla I have started - I figured you would not be on it since it has been 16 days.

**Comment by SushantChandla** [comment]
Okay, no problem : )

**Comment by creativecreatorormaybenot** [comment]
So it turns out, I only found a single place missing the doc (i.e. the abstract class `ProgressIndicator` docs) → so that was an easy fix.

I thought about adding an assertion for the value to be between 0.0 and 1.0, however, that would be a breaking change. Instead, I checked the code and it turns out the value is always clamped using `num.clamp`. That is why I added that to the `value` docs :)

The PR is #88800 🚀
